libkpod/attach.go

```diff
@@ -0,0 +1,182 @@
+package libkpod
+
+import (
+	"fmt"
+	"github.com/docker/docker/pkg/term"
+	"github.com/kubernetes-incubator/cri-o/oci"
+	"github.com/kubernetes-incubator/cri-o/utils"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	"golang.org/x/sys/unix"
+	"io"
+	"k8s.io/client-go/tools/remotecommand"
+	kubecontainer "k8s.io/kubernetes/pkg/kubelet/container"
+	"net"
+	"os"
+	"path/filepath"
+	"strconv"
+)
+
+/* Sync with stdpipe_t in conmon.c */
+const (
+	AttachPipeStdin  = 1
+	AttachPipeStdout = 2
+	AttachPipeStderr = 3
+)
+
+// ContainerAttach attaches to a running container
+// keys are string representation of what to use to detach
+// key format is a single character [a-Z] or ctrl-<value> where <value> is one of: a-z, @, ^, [, , or _.
+func (c *ContainerServer) ContainerAttach(container string, noStdIn bool, keys string) error {
+	// Check the validity of the provided keys first
+	var err error
+	detachKeys := []byte{}
+	if len(keys) > 0 {
+		detachKeys, err = term.ToBytes(keys)
+		if err != nil {
+			return errors.Wrapf(err, "invalid detach keys")
+		}
+	}
+
+	ctr, err := c.LookupContainer(container)
+	if err != nil {
+		return errors.Wrapf(err, "failed to find container %s", container)
+	}
+
+	cStatus := c.runtime.ContainerStatus(ctr)
+	if !(cStatus.Status == oci.ContainerStateRunning || cStatus.Status == oci.ContainerStateCreated) {
+		return errors.Errorf("%s is not created or running", container)
+	}
+
+	resize := make(chan remotecommand.TerminalSize)
+
+	AttachContainerSocket(ctr, resize, noStdIn, detachKeys)
+	close(resize)
+	c.ContainerStateToDisk(ctr)
+
+	return nil
+}
+
+// getStandardStreams returns the standard IO readers/writers
+func getStandardStreams() (stdIn, stdOut, stdErr *os.File) {
```

Comment:
I don't really see any reason for this function.  It is only used once so just initialize those values below.

Reply:
fixed